FeedSX/screens/createPost/index.tsx

```diff
@@ -0,0 +1,549 @@
+          pageCount: 0, // You need to specify the page count.
+          url: item.uri,
+        },
+        attachmentType:
+          item.type === 'image/jpg'
+            ? IMAGE_ATTACHMENT_TYPE
+            : item.type === 'video/mp4'
+            ? VIDEO_ATTACHMENT_TYPE
+            : 0, // You need to specify the attachment type.
+      };
+    });
+    setFormattedMediaAttachments([
+      ...formattedMediaAttachments,
+      ...convertedData,
+    ]);
+    return convertedData;
+  }
+
+  // convert the default selected document format to LMAttachmentUI
+  function convertDocData(data: docDefaultFormat[]): LMAttachmentUI[] {
+    const convertedData = data.map(item => {
+      return {
+        attachmentMeta: {
+          entityId: '', // You need to specify the entity ID.
+          format: item.type,
+          name: item.name,
+          ogTags: {
+            description: '',
+            title: '',
+            url: '',
+            image: '',
+          },
+          size: item.size,
+          duration: 0, // You need to specify the duration.
+          pageCount: 0, // You need to specify the page count.
+          url: item.uri,
+        },
+        attachmentType:
+          item.type === 'application/pdf' ? DOCUMENT_ATTACHMENT_TYPE : 0, // You need to specify the attachment type.
+      };
+    });
+    setFormattedDocAttachments([...formattedDocAttachments, ...convertedData]);
+    return convertedData;
+  }
+
+  // convert the LMOGTagsUI to LMAttachmentUI
+  function convertLinkData(data: LMOGTagsUI[]): LMAttachmentUI[] {
+    const convertedData = data.map(item => {
+      return {
+        attachmentMeta: {
+          entityId: '', // You need to specify the entity ID.
+          format: '',
+          name: '',
+          ogTags: {
+            description: item.description,
+            title: item.title,
+            url: item.url,
+            image: item.image,
+          },
+          size: 0,
+          duration: 0, // You need to specify the duration.
+          pageCount: 0, // You need to specify the page count.
+          url: '',
+        },
+        attachmentType: LINK_ATTACHMENT_TYPE, // You need to specify the attachment type.
+      };
+    });
+    setFormattedLinkAttachments([
+      ...formattedLinkAttachments,
+      ...convertedData,
+    ]);
+    return convertedData;
+  }
+
+  //select Images and videoes From Gallery
+  const selectGalley = async (type: string) => {
+    const options = {
+      mediaType: type,
+      selectionLimit: 0,
+    };
+    await launchImageLibrary(options as any, async (response: any) => {
+      if (response?.didCancel) {
+        console.log('didCancel', response.didCancel);
+      }
+      let selectedImages = response?.assets; // selectedImages can be anything images or videos or both
+
+      for (let i = 0; i < selectedImages?.length; i++) {
+        if (selectedImages[i].fileSize >= MAX_FILE_SIZE) {
```

Comment:
move this function to a util file, this code should not be present inside a view code and rename `selectGallery` to `selectImageVideo`

Reply:
resolved